From e6b33c857b67000e072ef18ff2f0d6a7b3e24157 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Wed, 12 Jun 2024 17:23:34 +0800
Subject: [PATCH 01/13] Revert "fix[drm]: Fix permission check issue on GKI
 kernel"

Create drm dumb failed on renderD128.

```
 mpp[20669]: mpp_drm: os_allocator_drm_alloc drm_alloc failed ret -1
 mpp[20669]: mpp_buffer: mpp_buffer_create failed to create buffer with size 397312
 mpp[20669]: hal_h265d_vdpu34x: h265d mpp_buffer_get failed
 mpp[20669]: mpp_hal: mpp_hal_init hal h265d_rkdec init failed ret -1
 mpp[20669]: mpp_hal: mpp_hal_init could not found coding type 16777220
 mpp[20669]: mpp_dec: mpp_dec_init could not init hal
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp_time: mpp_clock_put invalid clock (nil)
 mpp[20669]: mpp: error found on mpp initialization
 mpp[20669]: mpp_buffer: ~MppBufferService cleaning leaked group
 ffmpeg[20669]: mpp_mem_pool: put_pool found 4 used buffer size 224
```

This reverts commit 9c92ab9e20ba33bc9a663ec3b69d3be7c3ed558c.
---
 osal/allocator/allocator_drm.c | 21 ++++-----------------
 osal/mpp_runtime.cpp           |  4 +---
 2 files changed, 5 insertions(+), 20 deletions(-)

diff --git a/osal/allocator/allocator_drm.c b/osal/allocator/allocator_drm.c
index 6b1a07902..2e8566c9b 100644
--- a/osal/allocator/allocator_drm.c
+++ b/osal/allocator/allocator_drm.c
@@ -69,11 +69,7 @@ typedef struct {
     RK_U32  flags;
 } allocator_ctx_drm;
 
-/* use renderD128 first to avoid GKI kernel permission issue */
-static const char *dev_drm[] = {
-    "/dev/dri/renderD128",
-    "/dev/dri/card0",
-};
+static const char *dev_drm = "/dev/dri/card0";
 
 static RK_U32 to_rockchip_gem_mem_flag(RK_U32 flags)
 {
@@ -168,9 +164,8 @@ static int drm_free(int fd, RK_U32 handle)
 
 static MPP_RET os_allocator_drm_open(void **ctx, size_t alignment, MppAllocFlagType flags)
 {
-    allocator_ctx_drm *p;
     RK_S32 fd;
-    RK_S32 i;
+    allocator_ctx_drm *p;
 
     if (NULL == ctx) {
         mpp_err_f("does not accept NULL input\n");
@@ -181,17 +176,9 @@ static MPP_RET os_allocator_drm_open(void **ctx, size_t alignment, MppAllocFlagT
 
     mpp_env_get_u32("drm_debug", &drm_debug, 0);
 
-    for (i = 0; i < (RK_S32)MPP_ARRAY_ELEMS(dev_drm); i++) {
-        fd = open(dev_drm[i], O_RDWR | O_CLOEXEC);
-        if (fd > 0)
-            break;
-    }
-
+    fd = open(dev_drm, O_RDWR | O_CLOEXEC);
     if (fd < 0) {
-        mpp_err_f("open all drm device failed.\n");
-        mpp_err("Please check the following device path and access permission:\n");
-        for (i = 0; i < (RK_S32)MPP_ARRAY_ELEMS(dev_drm); i++)
-            mpp_err("%s\n", dev_drm[i]);
+        mpp_err_f("open %s failed!\n", dev_drm);
         return MPP_ERR_UNKNOW;
     } else {
         /* drop master by default to avoid becoming the drm master */
diff --git a/mpp/mpp.cpp b/mpp/mpp.cpp
index 32c14b848..eedeca7fa 100644
--- a/mpp/mpp.cpp
+++ b/mpp/mpp.cpp
@@ -50,18 +50,6 @@ static void mpp_notify_by_buffer_group(void *arg, void *group)
 
 static void *list_wraper_packet(void *arg)
 {
-    MppPacket packet = *(MppPacket*)arg;
-
-    if (mpp_packet_has_meta(packet)) {
-        MppMeta meta = mpp_packet_get_meta(packet);
-        MppFrame frm = NULL;
-
-        if (MPP_OK == mpp_meta_get_frame(meta, KEY_INPUT_FRAME, &frm)) {
-            mpp_assert(frm);
-            mpp_frame_deinit(&frm);
-        }
-    }
-
     mpp_packet_deinit((MppPacket *)arg);
     return NULL;
 }
@@ -229,8 +217,10 @@ MPP_RET Mpp::init(MppCtxType type, MppCodingType coding)
         mInitDone = 1;
     } break;
     case MPP_CTX_ENC : {
+        mPktIn  = new mpp_list(list_wraper_packet);
         mPktOut = new mpp_list(list_wraper_packet);
-        mFrmIn  = new mpp_list(list_wraper_frame);
+        mFrmIn  = new mpp_list(NULL);
+        mFrmOut = new mpp_list(NULL);
 
         if (mInputTimeout == MPP_POLL_BUTT)
             mInputTimeout = MPP_POLL_BLOCK;

From 697ad616175fbc6f0f0f24e0bba8a6d66aab3520 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Tue, 9 Apr 2024 18:22:19 +0800
Subject: [PATCH 03/13] fix[h264d]: Update hor align to 64 for high10 and rga3
 use case

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 mpp/hal/rkdec/h264d/hal_h264d_rkv_reg.c | 2 +-
 mpp/hal/rkdec/h264d/hal_h264d_vdpu34x.c | 2 +-
 mpp/hal/rkdec/h264d/hal_h264d_vdpu382.c | 2 +-
 3 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/mpp/hal/rkdec/h264d/hal_h264d_rkv_reg.c b/mpp/hal/rkdec/h264d/hal_h264d_rkv_reg.c
index 519008cd4..4e36bd473 100644
--- a/mpp/hal/rkdec/h264d/hal_h264d_rkv_reg.c
+++ b/mpp/hal/rkdec/h264d/hal_h264d_rkv_reg.c
@@ -229,7 +229,7 @@ static RK_U32 rkv_ver_align(RK_U32 val)
 
 static RK_U32 rkv_hor_align(RK_U32 val)
 {
-    return MPP_ALIGN(val, 16);
+    return MPP_ALIGN(val, 64); // align to 64 for h264 high10
 }
 
 static RK_U32 rkv_hor_align_256_odds(RK_U32 val)
diff --git a/mpp/hal/rkdec/h264d/hal_h264d_vdpu34x.c b/mpp/hal/rkdec/h264d/hal_h264d_vdpu34x.c
index d780bd786..3de838c76 100644
--- a/mpp/hal/rkdec/h264d/hal_h264d_vdpu34x.c
+++ b/mpp/hal/rkdec/h264d/hal_h264d_vdpu34x.c
@@ -329,7 +329,7 @@ static RK_U32 rkv_ver_align(RK_U32 val)
 
 static RK_U32 rkv_hor_align(RK_U32 val)
 {
-    return MPP_ALIGN(val, 16);
+    return MPP_ALIGN(val, 64); // align to 64 for h264 high10
 }
 
 static RK_U32 rkv_hor_align_256_odds(RK_U32 val)
diff --git a/mpp/hal/rkdec/h264d/hal_h264d_vdpu382.c b/mpp/hal/rkdec/h264d/hal_h264d_vdpu382.c
index bbc7f1673..79e1f5d02 100644
--- a/mpp/hal/rkdec/h264d/hal_h264d_vdpu382.c
+++ b/mpp/hal/rkdec/h264d/hal_h264d_vdpu382.c
@@ -335,7 +335,7 @@ static RK_U32 rkv_ver_align(RK_U32 val)
 
 static RK_U32 rkv_hor_align(RK_U32 val)
 {
-    return MPP_ALIGN(val, 16);
+    return MPP_ALIGN(val, 64); // align to 64 for h264 high10
 }
 
 static RK_U32 rkv_hor_align_256_odds(RK_U32 val)

From 06a3e44a285657c0e61d7e3daf17064508fa40ca Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Tue, 9 Apr 2024 18:24:53 +0800
Subject: [PATCH 04/13] fix[av1d]: Update hor align to 64 for main 10bit and
 rga3 use case

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 mpp/hal/vpu/av1d/hal_av1d_vdpu.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mpp/hal/vpu/av1d/hal_av1d_vdpu.c b/mpp/hal/vpu/av1d/hal_av1d_vdpu.c
index 5b6e1f9ad..c1e6edb11 100644
--- a/mpp/hal/vpu/av1d/hal_av1d_vdpu.c
+++ b/mpp/hal/vpu/av1d/hal_av1d_vdpu.c
@@ -115,7 +115,7 @@ static RK_U32 rkv_ver_align(RK_U32 val)
 
 static RK_U32 rkv_hor_align(RK_U32 val)
 {
-    return MPP_ALIGN(val, 16);
+    return MPP_ALIGN(val, 64); // align to 64 for av1 main 10bit
 }
 
 static RK_U32 rkv_len_align(RK_U32 val)

From 221c911f37c8a148fef1e544204d3134a9604b76 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Fri, 23 May 2025 22:56:05 +0800
Subject: [PATCH 05/13] fix[sys_cfg]: Update hor align to 64 for 10bit and rga3
 use case

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 mpp/base/mpp_sys_cfg.cpp | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/mpp/base/mpp_sys_cfg.cpp b/mpp/base/mpp_sys_cfg.cpp
index fe8725ce0..3fc1b1484 100644
--- a/mpp/base/mpp_sys_cfg.cpp
+++ b/mpp/base/mpp_sys_cfg.cpp
@@ -446,6 +446,7 @@ MPP_RET mpp_sys_dec_buf_chk_proc(MppSysDecBufChkCfg *cfg)
         aligned_height = cfg->v_stride ? cfg->v_stride : aligned_height;
 
         switch (type) {
+        case MPP_VIDEO_CodingAVC : /* ALIGN_64: for Hi10p/Hi422p + RGA3 use case */
         case MPP_VIDEO_CodingHEVC : {
             aligned_byte = mpp_sys_cfg_align(SYS_CFG_ALIGN_64, aligned_pixel_byte);
         } break;
@@ -457,8 +458,8 @@ MPP_RET mpp_sys_dec_buf_chk_proc(MppSysDecBufChkCfg *cfg)
                 aligned_byte = mpp_sys_cfg_align(SYS_CFG_ALIGN_256_ODD, aligned_pixel_byte);
         } break;
         case MPP_VIDEO_CodingAV1 : {
-            if (soc_type == ROCKCHIP_SOC_RK3588)
-                aligned_byte = mpp_sys_cfg_align(SYS_CFG_ALIGN_16, aligned_pixel_byte);
+            if (soc_type == ROCKCHIP_SOC_RK3588) /* ALIGN_64: for 10bit Main + RGA3 use case */
+                aligned_byte = mpp_sys_cfg_align(SYS_CFG_ALIGN_64, aligned_pixel_byte);
             else
                 aligned_byte = mpp_sys_cfg_align(SYS_CFG_ALIGN_128, aligned_pixel_byte);
         } break;
@@ -472,7 +473,7 @@ MPP_RET mpp_sys_dec_buf_chk_proc(MppSysDecBufChkCfg *cfg)
          * NOTE: rk3576 use 128 odd plus 64 for all non jpeg format
          * all the other socs use 256 odd on larger than 1080p
          */
-        if ((aligned_byte > 1920 || soc_type == ROCKCHIP_SOC_RK3576 || soc_type == ROCKCHIP_SOC_RK3399)
+        if (((aligned_byte * 8 / depth) > 1920 || soc_type == ROCKCHIP_SOC_RK3576)
             && type != MPP_VIDEO_CodingMJPEG) {
             rk_s32 update = 0;
 

From 6e83db78f0078d37e10239776ab8d50b0e77c836 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Tue, 26 Sep 2023 23:21:26 +0800
Subject: [PATCH 06/13] Revert "fix[hal_vp9d]: not support fast mode for
 rk3588"

This reverts commit 0d6cc57d44e79f8521ac2fc117571edb3224564b.
---
 mpp/hal/rkdec/vp9d/hal_vp9d_api.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mpp/hal/rkdec/vp9d/hal_vp9d_api.c b/mpp/hal/rkdec/vp9d/hal_vp9d_api.c
index e02039d78..2233a06ea 100644
--- a/mpp/hal/rkdec/vp9d/hal_vp9d_api.c
+++ b/mpp/hal/rkdec/vp9d/hal_vp9d_api.c
@@ -59,7 +59,7 @@ MPP_RET hal_vp9d_init(void *ctx, MppHalCfg *cfg)
         p->api = &hal_vp9d_vdpu34x;
         cfg->support_fast_mode = 1;
         if (mpp_get_soc_type() == ROCKCHIP_SOC_RK3588)
-            cfg->support_fast_mode = 0;
+            cfg->cfg->status.hal_task_count = 2;
     } else {
         p->api = &hal_vp9d_rkv;
         cfg->support_fast_mode = 0;

From 092b0e33bde15467503bb582a856c5853cd2575b Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Fri, 17 Jan 2025 11:28:56 +0800
Subject: [PATCH 07/13] fix[allocator]: Fix validating dma_heap drivers

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 osal/mpp_runtime.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/mpp/mpp.cpp b/mpp/mpp.cpp
index eedeca7fa..e6a9702dd 100644
--- a/mpp/mpp.cpp
+++ b/mpp/mpp.cpp
@@ -825,7 +825,8 @@ MPP_RET Mpp::get_packet(MppPacket *packet)
         MppBuffer buf = impl->buffer;
         RK_U32 offset = (RK_U32)((char *)impl->pos - (char *)impl->data);
 
-        mpp_buffer_sync_ro_partial_begin(buf, offset, impl->length);
+        if (buf)
+            mpp_buffer_sync_ro_partial_begin(buf, offset, impl->length);
 
         mpp_dbg_pts("%p output packet pts %lld\n", this, impl->pts);
     }
@@ -902,7 +903,8 @@ MPP_RET Mpp::get_packet_async(MppPacket *packet)
 
         impl = (MppPacketImpl *)pkt;
         offset = (RK_U32)((char *)impl->pos - (char *)impl->data);
-        mpp_buffer_sync_ro_partial_begin(impl->buffer, offset, impl->length);
+        if (impl->buffer)
+            mpp_buffer_sync_ro_partial_begin(impl->buffer, offset, impl->length);
     } else {
         AutoMutex autoFrameLock(mFrmIn->mutex());
 

From ff1c2c01a344016841d1cdb581b93d4d4067fef8 Mon Sep 17 00:00:00 2001
From: Brett Jia <dev.bjia56@gmail.com>
Date: Wed, 19 Jun 2024 11:44:23 -0400
Subject: [PATCH 09/13] fix[vp9d]: namespace vars that conflict with vpx

---
 mpp/hal/rkdec/vp9d/hal_vp9d_com.c | 30 +++++++++++++++---------------
 mpp/hal/rkdec/vp9d/hal_vp9d_com.h |  6 +++---
 2 files changed, 18 insertions(+), 18 deletions(-)

diff --git a/mpp/hal/rkdec/vp9d/hal_vp9d_com.c b/mpp/hal/rkdec/vp9d/hal_vp9d_com.c
index 340fb0f1e..3ef67bf62 100644
--- a/mpp/hal/rkdec/vp9d/hal_vp9d_com.c
+++ b/mpp/hal/rkdec/vp9d/hal_vp9d_com.c
@@ -838,7 +838,7 @@ static const RK_U8 vp9_default_coef_probs[4][2][2][6][6][3] = {
     }
 };
 
-const vp9_prob vp9_kf_y_mode_prob[INTRA_MODES][INTRA_MODES][INTRA_MODES - 1] = {
+const vp9_prob mpp_vp9_kf_y_mode_prob[INTRA_MODES][INTRA_MODES][INTRA_MODES - 1] = {
     {
         // above = dc
         { 137,  30,  42, 148, 151, 207,  70,  52,  91 },  // left = dc
@@ -953,7 +953,7 @@ const vp9_prob vp9_kf_y_mode_prob[INTRA_MODES][INTRA_MODES][INTRA_MODES - 1] = {
     }
 };
 
-const vp9_prob vp9_kf_uv_mode_prob[INTRA_MODES][INTRA_MODES - 1] = {
+const vp9_prob mpp_vp9_kf_uv_mode_prob[INTRA_MODES][INTRA_MODES - 1] = {
     { 144,  11,  54, 157, 195, 130,  46,  58, 108 },  // y = dc
     { 118,  15, 123, 148, 131, 101,  44,  93, 131 },  // y = v
     { 113,  12,  23, 188, 226, 142,  26,  32, 125 },  // y = h
@@ -966,7 +966,7 @@ const vp9_prob vp9_kf_uv_mode_prob[INTRA_MODES][INTRA_MODES - 1] = {
     { 102,  19,  66, 162, 182, 122,  35,  59, 128 }   // y = tm
 };
 
-const vp9_prob vp9_kf_partition_probs[PARTITION_CONTEXTS][PARTITION_TYPES - 1] = {
+const vp9_prob mpp_vp9_kf_partition_probs[PARTITION_CONTEXTS][PARTITION_TYPES - 1] = {
     // 8x8 -> 4x4
     { 158,  97,  94 },  // a/l both not split
     {  93,  24,  99 },  // a split, l not split
@@ -1013,8 +1013,8 @@ MPP_RET hal_vp9d_output_probe(void *buf, void *dxva)
     memset(buf, 0, 304 * 8);
 
     if (intraFlag) {
-        memcpy(partition_probs, vp9_kf_partition_probs, sizeof(partition_probs));
-        memcpy(uv_mode_prob, vp9_kf_uv_mode_prob, sizeof(uv_mode_prob));
+        memcpy(partition_probs, mpp_vp9_kf_partition_probs, sizeof(partition_probs));
+        memcpy(uv_mode_prob, mpp_vp9_kf_uv_mode_prob, sizeof(uv_mode_prob));
     } else {
         memcpy(partition_probs, pic_param->prob.partition, sizeof(partition_probs));
         memcpy(uv_mode_prob, pic_param->prob.uv_mode, sizeof(uv_mode_prob));
@@ -1073,11 +1073,11 @@ MPP_RET hal_vp9d_output_probe(void *buf, void *dxva)
             }
 
         //intra mode prob  80 x 128 bit
-        for (i = 0; i < INTRA_MODES; i++) { //vp9_kf_y_mode_prob
+        for (i = 0; i < INTRA_MODES; i++) { //mpp_vp9_kf_y_mode_prob
             RK_S32 byte_count = 0;
             for (j = 0; j < INTRA_MODES; j++)
                 for (k = 0; k < INTRA_MODES - 1; k++) {
-                    mpp_put_bits(&bp, vp9_kf_y_mode_prob[i][j][k], 8);
+                    mpp_put_bits(&bp, mpp_vp9_kf_y_mode_prob[i][j][k], 8);
                     byte_count++;
                     if (byte_count == 27) {
                         byte_count = 0;
@@ -1087,7 +1087,7 @@ MPP_RET hal_vp9d_output_probe(void *buf, void *dxva)
                 }
             if (i < 4) {
                 for (m = 0; m < (i < 3 ? 23 : 21); m++)
-                    mpp_put_bits(&bp, ((vp9_prob *)(&vp9_kf_uv_mode_prob[0][0]))[i * 23 + m], 8);
+                    mpp_put_bits(&bp, ((vp9_prob *)(&mpp_vp9_kf_uv_mode_prob[0][0]))[i * 23 + m], 8);
                 for (; m < 23; m++)
                     mpp_put_bits(&bp, 0, 8);
             } else {
@@ -1461,7 +1461,7 @@ MPP_RET hal_vp9d_prob_flag_delta(void *buf, void *dxva)
     memset(buf, 0, PROB_SIZE);
 
     if (intraFlag) {
-        memcpy(partition_probs, vp9_kf_partition_probs, sizeof(partition_probs));
+        memcpy(partition_probs, mpp_vp9_kf_partition_probs, sizeof(partition_probs));
     } else {
         memcpy(partition_flag, prob_flag->partition, sizeof(partition_flag));
         memcpy(partition_delta, prob_delta->partition, sizeof(partition_delta));
@@ -1572,11 +1572,11 @@ MPP_RET hal_vp9d_prob_flag_delta(void *buf, void *dxva)
             }
 
         //intra mode prob  80 x 128 bit
-        for (i = 0; i < INTRA_MODES; i++) { //vp9_kf_y_mode_prob
+        for (i = 0; i < INTRA_MODES; i++) { //mpp_vp9_kf_y_mode_prob
             RK_S32 byte_count = 0;
             for (j = 0; j < INTRA_MODES; j++)
                 for (k = 0; k < INTRA_MODES - 1; k++) {
-                    mpp_put_bits(&bp, vp9_kf_y_mode_prob[i][j][k], 8);
+                    mpp_put_bits(&bp, mpp_vp9_kf_y_mode_prob[i][j][k], 8);
                     byte_count++;
                     if (byte_count == 27) {
                         byte_count = 0;
@@ -1586,7 +1586,7 @@ MPP_RET hal_vp9d_prob_flag_delta(void *buf, void *dxva)
                 }
             if (i < 4) {
                 for (m = 0; m < (i < 3 ? 23 : 21); m++)
-                    mpp_put_bits(&bp, ((vp9_prob *)(&vp9_kf_uv_mode_prob[0][0]))[i * 23 + m], 8);
+                    mpp_put_bits(&bp, ((vp9_prob *)(&mpp_vp9_kf_uv_mode_prob[0][0]))[i * 23 + m], 8);
                 for (; m < 23; m++)
                     mpp_put_bits(&bp, 0, 8);
             } else {
@@ -1927,7 +1927,7 @@ MPP_RET hal_vp9d_prob_kf(void *buf)
 
     for (i = 0; i < 16; i++)//kf_partition_prob
         for (j = 0; j < 3; j++)
-            mpp_put_bits(&bp, vp9_kf_partition_probs[i][j], 8);
+            mpp_put_bits(&bp, mpp_vp9_kf_partition_probs[i][j], 8);
 
     //intra mode prob  80 x 128 bit, 10x 8x 128bit
     for (i = 0; i < 10; i++) {
@@ -1935,7 +1935,7 @@ MPP_RET hal_vp9d_prob_kf(void *buf)
         RK_U8 val = 0;
         for (j = 0; j < 10; j++)
             for (k = 0; k < 9; k++) {
-                mpp_put_bits(&bp, vp9_kf_y_mode_prob[i][j][k], 8);
+                mpp_put_bits(&bp, mpp_vp9_kf_y_mode_prob[i][j][k], 8);
                 byte_count++;
                 if (byte_count == 27) {
                     byte_count = 0;
@@ -1945,7 +1945,7 @@ MPP_RET hal_vp9d_prob_kf(void *buf)
         // 6x128 bit, 23x4 bytes
         if (i < 4) {
             for (m = 0; m < (i < 3 ? 23 : 21); m++) {
-                val = ((RK_U8 *)(&vp9_kf_uv_mode_prob[0][0]))[i * 23 + m];
+                val = ((RK_U8 *)(&mpp_vp9_kf_uv_mode_prob[0][0]))[i * 23 + m];
                 mpp_put_bits(&bp, val, 8);
             }
             for (; m < 23; m++)
diff --git a/mpp/hal/rkdec/vp9d/hal_vp9d_com.h b/mpp/hal/rkdec/vp9d/hal_vp9d_com.h
index 8dee8e569..3c41a9357 100644
--- a/mpp/hal/rkdec/vp9d/hal_vp9d_com.h
+++ b/mpp/hal/rkdec/vp9d/hal_vp9d_com.h
@@ -77,9 +77,9 @@ do{\
 }} while (0)
 
 
-extern const vp9_prob vp9_kf_y_mode_prob[INTRA_MODES][INTRA_MODES][INTRA_MODES - 1];
-extern const vp9_prob vp9_kf_uv_mode_prob[INTRA_MODES][INTRA_MODES - 1];
-extern const vp9_prob vp9_kf_partition_probs[PARTITION_CONTEXTS][PARTITION_TYPES - 1];
+extern const vp9_prob mpp_vp9_kf_y_mode_prob[INTRA_MODES][INTRA_MODES][INTRA_MODES - 1];
+extern const vp9_prob mpp_vp9_kf_uv_mode_prob[INTRA_MODES][INTRA_MODES - 1];
+extern const vp9_prob mpp_vp9_kf_partition_probs[PARTITION_CONTEXTS][PARTITION_TYPES - 1];
 
 #ifdef __cplusplus
 extern "C" {

From 22bf13b5f290542458e3b3128afdb7e6cf223536 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Fri, 23 May 2025 23:11:14 +0800
Subject: [PATCH 10/13] fix[buf_slot]: Disable sys_cfg_info warnings for now

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 mpp/base/mpp_buf_slot.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/mpp/base/mpp_buf_slot.cpp b/mpp/base/mpp_buf_slot.cpp
index 7a0d17a21..1683e05fc 100644
--- a/mpp/base/mpp_buf_slot.cpp
+++ b/mpp/base/mpp_buf_slot.cpp
@@ -493,6 +493,7 @@ static void generate_info_set(MppBufSlotsImpl *impl, MppFrame frame, RK_U32 forc
 
     if (impl->align_chk_log_en) {
         impl->align_chk_log_en = 0;
+#if 0
         if (legacy_info_set.h_stride_by_pixel != sys_cfg_info_set.h_stride_by_pixel)
             mpp_logi("mismatch h_stride_by_pixel %d - %d\n",
                      legacy_info_set.h_stride_by_pixel,
@@ -509,6 +510,7 @@ static void generate_info_set(MppBufSlotsImpl *impl, MppFrame frame, RK_U32 forc
             mpp_logi("mismatch size_total %d - %d\n",
                      legacy_info_set.size_total,
                      sys_cfg_info_set.size_total);
+#endif
     }
 }
 

From 7333dec7c04efbe27687268bea981d06586bebc0 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Fri, 14 Mar 2025 00:54:30 +0800
Subject: [PATCH 11/13] Revert "fix[os_log]: Modify default log option for
 linux"

This reverts commit 4ed4105a3c4502d9e63fbe8c6c138c8356329958.
---
 osal/linux/os_log.cpp | 11 +++--------
 1 file changed, 3 insertions(+), 8 deletions(-)

diff --git a/mpp/CMakeLists.txt b/mpp/CMakeLists.txt
index 7cbfecc14..d796ec30e 100644
--- a/mpp/CMakeLists.txt
+++ b/mpp/CMakeLists.txt
@@ -49,15 +49,19 @@ set (MPP_SRC
 set(MPP_VERSION "0")
 set(MPP_ABI_VERSION "1")
 
-add_library(${MPP_SHARED} SHARED ${MPP_SRC})
-target_link_libraries(${MPP_SHARED} ${ASAN_LIB})
-set_target_properties(${MPP_SHARED} PROPERTIES C_VISIBILITY_PRESET default)
-set_target_properties(${MPP_SHARED} PROPERTIES CXX_VISIBILITY_PRESET default)
+if(BUILD_SHARED_LIBS)
+    add_library(${MPP_SHARED} SHARED ${MPP_SRC})
+    target_link_libraries(${MPP_SHARED} ${ASAN_LIB})
+    set_target_properties(${MPP_SHARED} PROPERTIES C_VISIBILITY_PRESET default)
+    set_target_properties(${MPP_SHARED} PROPERTIES CXX_VISIBILITY_PRESET default)
 
-# NOTE: due to legacy libray naming issue we can not support version on Android
-if (NOT ANDROID)
-set_target_properties(${MPP_SHARED} PROPERTIES VERSION ${MPP_VERSION})
-set_target_properties(${MPP_SHARED} PROPERTIES SOVERSION ${MPP_ABI_VERSION})
+    # NOTE: due to legacy libray naming issue we can not support version on Android
+    if (NOT ANDROID)
+    set_target_properties(${MPP_SHARED} PROPERTIES VERSION ${MPP_VERSION})
+    set_target_properties(${MPP_SHARED} PROPERTIES SOVERSION ${MPP_ABI_VERSION})
+    endif()
+else()
+    add_library(${MPP_SHARED} STATIC ${MPP_SRC})
 endif()
 
 set_target_properties(${MPP_SHARED} PROPERTIES FOLDER "mpp")
@@ -65,22 +69,17 @@ set_target_properties(${MPP_SHARED} PROPERTIES CLEAN_DIRECT_OUTPUT 1)
 target_link_libraries(${MPP_SHARED} mpp_codec mpp_hal mpp_vproc kmpp
                       ${BEGIN_WHOLE_ARCHIVE} mpp_base kmpp_base osal ${END_WHOLE_ARCHIVE})
 
-# build static library
-add_library(${MPP_STATIC} STATIC ${MPP_SRC})
-set_target_properties(${MPP_STATIC} PROPERTIES FOLDER "mpp" OUTPUT_NAME "${MPP_SHARED}")
-set_target_properties(${MPP_STATIC} PROPERTIES CLEAN_DIRECT_OUTPUT 1)
-target_link_libraries(${MPP_STATIC} mpp_codec mpp_hal mpp_vproc kmpp
-                      ${BEGIN_WHOLE_ARCHIVE} mpp_base kmpp_base osal ${END_WHOLE_ARCHIVE})
-
-add_custom_command(TARGET ${MPP_STATIC} POST_BUILD
-    COMMAND ${CMAKE_AR} x $<TARGET_FILE:${MPP_STATIC}>
-    COMMAND ${CMAKE_AR} rcs lib${MPP_SHARED}.a *.o
-    COMMAND ${CMAKE_STRIP} --strip-debug lib${MPP_SHARED}.a
-    COMMAND pwd
-    COMMAND rm *.o
-    )
 
 add_subdirectory(legacy)
 
-install(TARGETS ${MPP_SHARED} LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-install(TARGETS ${MPP_STATIC} ARCHIVE DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+if(BUILD_SHARED_LIBS)
+    install(TARGETS ${MPP_SHARED} LIBRARY DESTINATION "${CMAKE_INSTALL_LIBDIR}")
+else()
+    add_custom_command(TARGET ${MPP_SHARED} POST_BUILD
+    COMMAND ${CMAKE_SOURCE_DIR}/merge_static_lib.sh ${CMAKE_BINARY_DIR} ${MPP_SHARED}_merged
+    COMMAND ${CMAKE_STRIP} --strip-debug ${CMAKE_BINARY_DIR}/mpp/lib${MPP_SHARED}.a
+    COMMENT "Building a merged static lib."
+    )
+
+    install(FILES ${CMAKE_BINARY_DIR}/mpp/lib${MPP_SHARED}_merged.a TYPE LIB RENAME lib${MPP_SHARED}.a)
+endif()

From 48fb6aa79c8b48e1ca98ced18233fcd8a6ac68c5 Mon Sep 17 00:00:00 2001
From: nyanmisaka <nst799610810@gmail.com>
Date: Sat, 24 May 2025 01:23:51 +0800
Subject: [PATCH 13/13] fix[vproc]: Fix unit tests cannot be disabled

Signed-off-by: nyanmisaka <nst799610810@gmail.com>
---
 mpp/vproc/iep/test/CMakeLists.txt  | 10 ++++++----
 mpp/vproc/iep2/test/CMakeLists.txt | 12 +++++++-----
 mpp/vproc/rga/test/CMakeLists.txt  | 10 ++++++----
 mpp/vproc/vdpp/test/CMakeLists.txt | 22 ++++++++++++----------
 4 files changed, 31 insertions(+), 23 deletions(-)

diff --git a/mpp/vproc/iep/test/CMakeLists.txt b/mpp/vproc/iep/test/CMakeLists.txt
index 5f3069f55..ceb55f943 100644
--- a/mpp/vproc/iep/test/CMakeLists.txt
+++ b/mpp/vproc/iep/test/CMakeLists.txt
@@ -4,7 +4,9 @@
 # ----------------------------------------------------------------------------
 # iep unit test
 option(IEP_TEST "Build base iep unit test" ${BUILD_TEST})
-add_executable(iep_test iep_test.cpp)
-target_link_libraries(iep_test ${MPP_SHARED} utils)
-set_target_properties(iep_test PROPERTIES FOLDER "mpp/vproc/iep")
-add_test(NAME iep_test COMMAND iep_test)
+if (IEP_TEST)
+    add_executable(iep_test iep_test.cpp)
+    target_link_libraries(iep_test ${MPP_SHARED} utils)
+    set_target_properties(iep_test PROPERTIES FOLDER "mpp/vproc/iep")
+    add_test(NAME iep_test COMMAND iep_test)
+endif()
diff --git a/mpp/vproc/iep2/test/CMakeLists.txt b/mpp/vproc/iep2/test/CMakeLists.txt
index e54298bc4..437738a42 100644
--- a/mpp/vproc/iep2/test/CMakeLists.txt
+++ b/mpp/vproc/iep2/test/CMakeLists.txt
@@ -3,8 +3,10 @@
 # mpp/vproc/iep2 built-in unit test case
 # ----------------------------------------------------------------------------
 # iep2 unit test
-option(IEP2_TEST "Build base iep2 unit test" ON)
-add_executable(iep2_test iep2_test.c)
-target_link_libraries(iep2_test ${MPP_SHARED} utils)
-set_target_properties(iep2_test PROPERTIES FOLDER "mpp/vproc/iep2")
-add_test(NAME iep2_test COMMAND iep2_test)
+option(IEP2_TEST "Build base iep2 unit test" ${BUILD_TEST})
+if (IEP2_TEST)
+    add_executable(iep2_test iep2_test.c)
+    target_link_libraries(iep2_test ${MPP_SHARED} utils)
+    set_target_properties(iep2_test PROPERTIES FOLDER "mpp/vproc/iep2")
+    add_test(NAME iep2_test COMMAND iep2_test)
+endif()
diff --git a/mpp/vproc/rga/test/CMakeLists.txt b/mpp/vproc/rga/test/CMakeLists.txt
index af0ee4360..a6ea356f5 100644
--- a/mpp/vproc/rga/test/CMakeLists.txt
+++ b/mpp/vproc/rga/test/CMakeLists.txt
@@ -4,7 +4,9 @@
 # ----------------------------------------------------------------------------
 # rga unit test
 option(RGA_TEST "Build base rga unit test" ${BUILD_TEST})
-add_executable(rga_test rga_test.cpp)
-target_link_libraries(rga_test ${MPP_SHARED} utils)
-set_target_properties(rga_test PROPERTIES FOLDER "mpp/vproc/rga")
-add_test(NAME rga_test COMMAND rga_test)
+if (RGA_TEST)
+    add_executable(rga_test rga_test.cpp)
+    target_link_libraries(rga_test ${MPP_SHARED} utils)
+    set_target_properties(rga_test PROPERTIES FOLDER "mpp/vproc/rga")
+    add_test(NAME rga_test COMMAND rga_test)
+endif()
diff --git a/mpp/vproc/vdpp/test/CMakeLists.txt b/mpp/vproc/vdpp/test/CMakeLists.txt
index a88935d6a..87c389881 100644
--- a/mpp/vproc/vdpp/test/CMakeLists.txt
+++ b/mpp/vproc/vdpp/test/CMakeLists.txt
@@ -3,14 +3,16 @@
 # mpp/vproc/vdpp built-in unit test case
 # ----------------------------------------------------------------------------
 # vdpp unit test
-option(VDPP_TEST "Build base vdpp unit test" ON)
-add_executable(vdpp_test vdpp_test.c)
-target_link_libraries(vdpp_test ${MPP_SHARED} utils vproc_vdpp)
-set_target_properties(vdpp_test PROPERTIES FOLDER "mpp/vproc/vdpp")
-add_test(NAME vdpp_test COMMAND vdpp_test)
+option(VDPP_TEST "Build base vdpp unit test" ${BUILD_TEST})
+if (VDPP_TEST)
+    add_executable(vdpp_test vdpp_test.c)
+    target_link_libraries(vdpp_test ${MPP_SHARED} utils vproc_vdpp)
+    set_target_properties(vdpp_test PROPERTIES FOLDER "mpp/vproc/vdpp")
+    add_test(NAME vdpp_test COMMAND vdpp_test)
 
-# hwpq test (call libvdpp.so)
-add_executable(hwpq_test hwpq_test.cpp)
-target_link_libraries(hwpq_test vdpp)
-set_target_properties(hwpq_test PROPERTIES FOLDER "mpp/vproc/vdpp")
-add_test(NAME hwpq_test COMMAND hwpq_test)
+    # hwpq test (call libvdpp.so)
+    add_executable(hwpq_test hwpq_test.cpp)
+    target_link_libraries(hwpq_test vdpp)
+    set_target_properties(hwpq_test PROPERTIES FOLDER "mpp/vproc/vdpp")
+    add_test(NAME hwpq_test COMMAND hwpq_test)
+endif()

